src/main/java/List.java

```diff
@@ -0,0 +1,221 @@
+public class List {
+    public static final String INVALID_EVENT_INPUT_MESSAGE = "event <event name> /from <start date/time> /end <end date/time>";
+    public static final String INVALID_DEADLINE_INPUT_MESSAGE = "deadline <deadline name> /by <deadline>";
+    public static final String INVALID_TODO_INPUT_MESSAGE = "todo <task name>";
+    public static final int DEADLINE_WORD_LEN = 8;
+    public static final int INPUT_SPACE_BUFFER = 2;
+    public static final int TODO_WORD_LEN = 4;
+    public static final String DEADLINE_BY_KEYWORD = "/by";
+    public static final String EVENT_FROM_KEYWORD = "/from";
+    public static final String EVENT_TO_KEYWORD = "/to";
+
+    private int numItems;
+    private Task[] itemList = new Task[100];
+
+    public List() {
+        numItems = 0;
+    }
+
+    public int getNumItems() {
+        return numItems;
+    }
+
+    public void addItem(String line) {
+        if (line.length() >= 7 && line.substring(0, 5).equals("event")) {
+            if (line.contains(EVENT_FROM_KEYWORD) && line.contains(EVENT_TO_KEYWORD)) {
+                addEvent(line);
+            } else {
+                printInvalidEventMessage();
+            }
+        } else if (line.length() >= (DEADLINE_WORD_LEN + INPUT_SPACE_BUFFER) && line.substring(0, DEADLINE_WORD_LEN).equals("deadline")) {
```

Comment:
Would it be better to break up this long expression into small, more bite-sized steps?

Reply:
Done thank you